tmt/steps/provision/testcloud.py

```diff
@@ -251,13 +253,14 @@ def options(cls, how=None):
                 '-a', '--arch',
                 type=click.Choice(['x86_64', 'aarch64', 's390x', 'ppc64le']),
                 help="What architecture to virtualize, host arch by default."),
-            ] + super().options(how)
+            ]
+        return options
 
-    def default(self, option, default=None):
+    def default(self, option: str, default: Any = None) -> Any:
         """ Return default data for given option """
         return getattr(TestcloudGuestData(), option, default)
 
-    def wake(self, keys=None, data=None):
+    def wake(self, keys: Optional[List[str]] = None, data: Any = None) -> None:
```

Comment:
@janhavlin data should be here `TestcloudGuestData` I do believe, like in https://github.com/teemtee/tmt/blob/462d064924859e37f2945d60585cedbd184c9478/tmt/steps/provision/artemis.py#L369

Reply:
I followed how super class [`ProvisionPlugin`](https://github.com/teemtee/tmt/blob/3f79f40f14c84b46c77f9933b98727f8b8313186/tmt/steps/provision/__init__.py#L240) defines the method so I went with `GuestData`. Is it okay?

When I used `TestcloudGuestData` here, I was getting:
```
tmt/steps/provision/testcloud.py:263: error: Argument 2 of "wake" is incompatible with supertype "ProvisionPlugin"; supertype defines the argument type as "Optional[GuestData]"  [override]
tmt/steps/provision/testcloud.py:263: note: This violates the Liskov substitution principle
tmt/steps/provision/testcloud.py:263: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
```

Reply:
Yes, that's expected. Use `TestcloudGUestData`, it's a correct and more specific type, which means mypy is correct, too. I hope to make mypy happy with https://github.com/teemtee/tmt/issues/1437.

Reply:
Alright so I will use `TestcloudGuestData` and use a `type: ignore` and add a TODO comment to revisit it once `Guest` becomes a generic type.